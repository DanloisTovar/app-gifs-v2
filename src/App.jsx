import './assets/styles/App.css';

function App() {
    return (
        <>
            <h1 className="text-4xl font-bold text-indigo-500">App Gift</h1>
        </>
    );
}

export default App;
