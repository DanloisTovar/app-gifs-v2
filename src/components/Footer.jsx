export const Footer = () => {
	return (
		<div className="bg-gray-400 text-white py-4 sm:py-6 md:py-8 lg:py-16">
			<div className="container mx-auto px-4 sm:px-6 md:px-8 lg:px-10">
				<div className="">
					<p className="text-center text-sm sm:text-base md:text-lg lg:text-xl">
						Â© 2023 DanloisDev. Todos los derechos reservados.
					</p>
				</div>
			</div>
		</div>
	);
};
