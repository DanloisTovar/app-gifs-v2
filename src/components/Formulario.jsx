import PropTypes from 'prop-types';
import { useState } from 'react';

export const Formulario = ({ onAddNewCategory }) => {
    // hooks:
    const [inputValue, setInputValue] = useState('');
    // events:
    const onHandleChange = ({ target }) => {
        setInputValue(target.value);
    };
    const onHandleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;

        onAddNewCategory(inputValue.trim());
        setInputValue('');
    };
    return (
        <div className="container flex items-center justify-center ">
            <form
                className="w-full sm:w-3/4 md:w-2/3 lg:w-1/2 xl:w-1/3 max-w-sm p-5"
                onSubmit={onHandleSubmit}
            >
                <div className="flex items-center border-b-2 border-indigo-500 py-2">
                    <input
                        className="appearance-none bg-transparent border-none w-full text-gray-700 mr-3 py-1 px-2 leading-tight focus:outline-none"
                        type="text"
                        value={inputValue}
                        placeholder="Â¿Que gif deseas?"
                        onChange={onHandleChange}
                    />
                    <button
                        className="flex-shrink-0 bg-indigo-500 hover:bg-indigo-700 border-indigo-500 hover:border-indigo-700 text-sm border-4 text-white py-1 px-2 rounded"
                        type="button"
                        onClick={onHandleSubmit}
                    >
                        Buscar
                    </button>
                </div>
            </form>
        </div>
    );
};
Formulario.propTypes = {
    onAddNewCategory: PropTypes.func.isRequired,
};
