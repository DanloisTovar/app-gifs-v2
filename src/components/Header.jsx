export const Header = () => {
    return (
        <div className="relative isolate overflow-hidden bg-gray-900 py-24 sm:py-32">
            <img
                src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxATEhUSEhIVFRUXFhUYFxUXFRUVFRgVFRUWFhUYFxcYHSggGB0lHRUVITEhJSkrLi4uFx8zODMtNygtLisBCgoKDg0OGhAQGy8lHyUtLS0tLS0tLS0tLS0tLS0tLy0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLf/AABEIAJoBSAMBIgACEQEDEQH/xAAbAAACAwEBAQAAAAAAAAAAAAABAgADBAUGB//EAEEQAAEDAQUECAQEAwcFAQAAAAEAAgMRBBIhMUEFUWFxBhMiMoGRscFCoeHwUmJy0RQjsgckM4KSwvE0Y3ODolP/xAAaAQACAwEBAAAAAAAAAAAAAAABAgADBAUG/8QANREAAQMDAQYDBwMEAwAAAAAAAQACEQMhMVEEEkFhcfAFIqETMoGRwdHhQrHxFCNSgiRikv/aAAwDAQACEQMRAD8A6z5VnkkVDpVU+VdmnQheUobDdGWRcnbEd+KRm9jgOdMPnRa5JFllkXRo0ZsV6PY9liF5j+z2jrQ6A5Wmz2iDkXsvN+bAuHbqgRNOYjFRuLnuePGjmq/Ztq/hrdG/uiKdh/yNkaT5tWrptZ7lvtDKUpIaD8pALP8A5IXmQ0s8Qc3VgP8Aswlp9HtnoFu/UuGEyYimGuvDglXTCsCitjjwvHL5k7h++iaGIUvOyy4uO4fKp08gUkkLjXwAGQG4KC6Ek4Qc6v7aBQBFja4Lu9DtkNtVqZE7uCr5CMKsbmBuqSG14qVajaTHVH4Ak/D66aoucGiSuh0T6Cz2sCRx6mHRxFXP33G7vzHDnivbx/2d7NYKOEkh/E6Sh8mABemdMGgNaAAAAAMAAMAANAsc1pXkRtHiG21N7eLG8GttA4Sck63jkF5/a/F92zSvG7Y/s8s5BNnkcx2jXkFnK81t4c8V8+2hYJIHmOVt1w8iNCDqF9lmtC8z0vsbZoSaduMFzTrQYuZ4j5gL1GxsrMbD3F3X75PxSeG+J7Q6ru1LtPzHPpyP8/NkaIhRboXqIXR6PbPknnZDGKl5o6vdDK1c53AAV5gUxoun0t2ix12zwH+7wtDYzh/MdUmSU0zLi0/Sqo6L7ajgMrJIy5k0Zje9huzMa7vdW44UOoOdBjgrto7DpGZIJBPATUPHejNK0maMWYYE5YA8FzyCds3qohrQNzQkjzOng4DyNBvulxbO95c1So1jxv200J64nQZPCSvOIJnCmGqi6MLShRLRMopCiRbK3e2c8AzmAGud8lVDDeNNNXflQtEl43vIfhar6Q9mPanOG9eJ/wBeH/aOAKHFUlBMUFmhRRAhROKHA+B/fglSpprS5yRkSeJuNDmtkcSRrFA1UxxLTHErY4k87wwb3HBrdSeSuDABJTwuLbe+770CoXZt2wrTGW9YzF4Dg6oujVwecmkarLPY2XSY33izCQUpSpoHMriWVN3GhrT8QAykzdZBtFNwBaZB4jGmetuvxjnkKJmu8t33qo9lOIOR+8iorZQpXmgonpXn68lFMKtRFBRFfWjKqnyrKZlTaLSGirnBg3kgeq7zaEXKWlsEXhaZJVllkXMtW37O3OS9+gE/PJcy09J2/BG536nU+VCj/UbPS994/c/ISujTbRpe84fus22LPW2MBylMVfG6w+i63TqW/JZ7SAK2myQPcR/+gDo5AP8AS1cQbQfLJfcACyOUilcP5brtSfzU8117Z/N2TZ341s9plgP6ZWidvoQvH7fUYfEWVmE7pcR/7a4n5vY30XOrOa+qXNwT+/5lebC0RRCl53d4ZuO5vudPIFI2jN3d+ZO4fvp5AiSQuOPIDQDcF0olGZMJpZS41NBoAMgNw+/mr9nOYJGl8fWNriypbeG6oxH3kszGEmgzV73Bout8Xb+A4eqYtEbv7Ej1EEdQQdEeQQmLRUNyrnWuGgB3cdV7H+y14E0x16rDlfZX/avErr9Gdo/w87ZD3SCx36XZnwIB8EK1IVKZYeKzbdSfU2Z7KfvRbqL/AML69NaFhltCzSWoHEGoWSW0J6GxgWAXmdl8Nc8y66vltCxTz4GuVFnltC4u3to3Yy0HtSVa3/cfAey6raAptL3YHffNep2XwwMbvOFgvKQYq0sS2dn3+ysc5c6ndoWpuFWttg2hLC5r4nuB1oTQiuRG7msiFFbAwVHta9pa4SDkG4XdlEFpxAEE2o7sTjw/AeC49os72OLXtLSND94hIFtZbiW3ZG32aV77f0n2TNY3EwhSpBnlBtzvHxzHeFgUurVLZxS+032eRH6m+6SI0F+n6B+b8XgrRQO9DjHGeWvPlqbK1zS0wVH9nsDPN/MYtCzUTpVXVfvnQDA0Gn31MnilUQRooqUEpCVMoRuSoFOwg4HDcd3A8PRWNtT2YHGm/wDcLOrmuDhddh+F27geHokkjCrPlurjtR+gb8/3Xc2FZWtf1zqlx7odiWc958uQyB2H0acAJphQ5saf6j7eavt9mb+Om/FuI1B+mK07IaNQkudJGBNp58PsrTR9qwhxyqekfSHrWmCKhBpfeaXRQ1wJyx18lwGWgAFrO+WlpkI7wdgWtB7tRhU4ngt7dlsA7MtWE17rJB50XKcGVNQLlSA4Xg51PwipHslr7K5sTAJ5iPhE25mJ9FlGxMpM3AOfX7jCWwWGSaVkMbSZHuDWt4nfuAzJ0AK9Rbeh4q4WO0MtQjN2dgq2Rjm4SPDMS9lb1C0E6UdjXTsqI2WzCcYWu1Nc2z1PajsxHbnccLrjkHYYY71woZnQ3SyR0TWn/FFRJI9uBuDA0GIocBU3sTRecfXrVahdReA1sgWkPMkE6wCN1u7cuDzBaEpJdjvvkscmzg7GIkgVvB1AQBm86BuVans1xJFHGmaytDbzJL5bS/QEUrk5tcS2uFSBjTeF6Rm3bPaTSb+7ykgidtA2QjAfxF0AXv8AuAUxOAFQeZtPZ0kElboxBNBUxSxkdoxkZtpm3MZjDLWx1X9Vo4Z+Zx0IAj9UqgbUA/2b7O4Tx6HBHqOIC41L3Pdv4jiorLTEGkFpJacWnXiDxH7HIhRaRdbAJwtElsnfnOaa0NwewJ4LOYXu7rS4D8Pbz1JbXE71a23AUDYY8Mr16T11RftW0OwEhHBtGf04rU8Uzd7y48gSPm6D6K1wabudJ6fcgqyLo9aXYlgYN73NYPma/JaI9i2dv+NbI2ndGHTEbwCKAHisRtUjML7i45m8TTgMc+Krba97GHmxo+bQD81kqNB90fM/YBYnsqPw6ByifVrl3LI7ZUbJHtjtU2DWkSPjiY6rg4AdWC4dyvghbukgfZ3WaKywwQuc1xDQ8yFzBh23Ox3VpkVxZLSC0tDGtBcCbt+pLQ4DvOP4iqa1VDdkpudvvBsQRJOREGAd2QRbKels4YLkk6kz+PRFz6+w0A3BRjSTQJQFZeoKDxPtyW1X9E5cALrdczv4Dh6pEEzRqfAfeicCEQIRA1PgN/0QrVQlFPCYLr7K29JELh7bNBXFvI7uC6jukUJzLhwLT7VXlQjSq0U6z2Y9fwQmZDTIAXatXSBmUbS9289ln7rhSyPeescaurnw0A4KwMGQ896PyKSrUq1oDzYcBjvrKepUfUgHGgUaeCCiIUaIslQCZSigTQilTBNRKUcKKNcQag0cmlkLs0KJSjvODS2baKIIUTKJFEihUKiQoIIVTFBKgVCNR4jd9EpRBUcNR4j70SlImDg7B3g72PD0XW2JsEyVll7MLTQnV7tGt36Zb6DHKrYuzWPrLO4x2dho53xPdSoiiBzeRno0YnQG/bXSF01I42iKJooxrfhblTxGZzNToaLDUrPfUNKiMe8ThvLm7UDAgnISCxshtjaLa3WgUGHV/DT85H9I8TUXRd0S2KLTMZZa/wAPEA+XeQD2Imje44ClMK5Lk7M2bLO8RxtvOJA3CrjQVOnvovSdJNpCyxjZ1mfVrD/eJW4GSc9mQA5hre7vwpzo2yrUMbPSd/cdk/4twXcjwZxkg3DSo8knmVj6S7XD5nvfRz3EAMBNxjW9yOo+BtB2BSru06h7I4U05mPaPayBwAI0buFNNNFlPyQKto7KygwMZwt6R+wA6ADghuWQIXU2ZtuSIdWQJIiamN2QNa1Yc2O4hYr1/A97Q/i4HjxVBCvykextRu68d6j7r009ghmaX2dxLTi6IjtscPjaB3hjQgaHfRRees87mGrSQeBoeYOhUTeTjPw/hVii5tmkrVNse0NzjPgWu9Csp7OHxa8PqvTWq3cVw7daGuzFfVVgkq/OVgUCJCATpkUyVFMEU9UEE4ThFQIlRAJkUUyAUCdFMrMBhTxUyy896km775IhFEuG4c0hKigCsARRCNEAmTIqI0QTJoTIKURogpCiiWiZRCEEqCJQISkIKJSigkKCChUKBSoJSmacf23IJSkKBWm1Wtz6A4NaKMYO60a0HHMnMnNNBZBQOkqAe6wd99cro0H5j4A5KyxNaGveGhz20wdS6G/ju/HQ0FDgKjA407nRvZ3WSvkncQ2JvWWiQ1qyMZsac+tf3Rq0Vp2sG4NprsoU3HAbpk9NJJF8kmBJc2anGAujC8WCyfxLgG2mW82zRjuwtu3XzY959DdvYnGgwXgnvqul0j2u60zukIDWgBsbBlHE3CNg5D5krllJsWzupsL6o87ru5aNB0be95cXOy4qNEZQSlMgtZRSq2t7PPfv58VUoggRKKCcmqiiKulnLkjWKxrFY1iZrUYSCKqzEUXSaxZLYyjuYBRc20qEKprSclF3th210VS2ASNe2443Re/NjqCdNwCy7X2W6P8AmNaTGT/prkD7JxSfue0i3eNR0SM9oZJbAGDP04eq5iIQoiEArEyCgVgFBVMilCKsa6uBCV7aFOEVGuoogiE4TKBMFAgmCiYIhEKJwmURAUCZMAiohRRFGJRSIFMUqUoIFApikSlKgiiUpSYQUQURa2pokKCUpFbepkES0EV1SlKVv2Sx4o+Nt+UvDGNHwl1ACRrerdGmetF2el0zbPE2ww0uhxfaXh169aKUdHXVsYIGOZxzCt6D2csitFpgpLamNc2GAEX2h12/PdOL6A0AFcSeC868Al5ocT22mpe11cX44uAJOJxxIOYJ41q+1uk2pkW1eRLZH+LAfJaC8k3IBOdzodOnf8a44rlqFWyxlpIOnkdQRwIxVRXTV0yggiglKiVRRRBBRRRRRRdJrFc1idrFe1i1NarAq2sXN2l36bgP3912iAASchiV56WS84u3n/hLVsAEHLXs62XHY4g4EVIBHh3TxGS27UMzG9mV74JMAb3mx4GF4eRGIXEWuz2pzQ5ubXChae6d3Ig4g+yzP33QQccDiPoeI4cDqlJKzgp7uuiSidjqe4VwUUWi7eA3+vDmqXs1GX3gVGPITBHKuDcapZzim60nLP1+vr61JgmCgTJEQrAmTBMggmUTIhQIpwmURCUIohFOgUKqIzKiCBUKKUqJVFClQlKoVFEpSGyChTQnFKokKCtuUrxTAXR94fVJ1pHP0+v3yqe8lIlymgmexwexxa4Goc0kEHgRkvSxbYhtVP4odXMMBamMaK4UpaG5PHEacKrzDGanL33BLI6vsFnq0WVCCcjBGRy6agyOSpq0m1LH5jI+nUEEagrt7X2RIwCoBFKsew3ontzPVP1GZAzGOYII4C6uyttTQdnvxk4xuxaeIHwniFqtWz4rReksnezfCSL43llMHN++CYA4Kz0nVKXkrXHBwx8Rcg/EtOowvPoJntIJBFCMwcwUqGVsSqKKIIKKIhRSDwUXo2sVpAAqTQb1yXbXdo0DnUrHPaHv7zq8NPJajWaMKwvC07St1/st7vr9FgC9HZuhNuc0PcxsMZFesnkZE2meRN75LVH0Yscbb9o2jEQDQizNfPU/ha/BtcRvXIf4psxfAfvHRgL/AIQ0GOhVJqCV5iKEuIDRUnTXf5cVq/hGULWvvSDGgpcIGbWn4nDOuRoaVwr327X2RCy7HZrRNj2jNK2EOx1ENSRl2ThhvQj6WTN/6SyWaCuF9lnHWDm99a86JW19qrn+zRd/sWt+YG++DpujrwC7zjgLk2TYdqkaXMs8paGl3WdW+5dAri4imX3uw3aZj967l7HY/SO1vtcD555XNL2xyRmRxZWUXAerHZob1csCDuC8zbY7sj4iaFjnsDj+R5bQ+WenLLRRftHtHMrgAiCIm1yCDIFwQOAylbW85YcxMcY/H1ESsjX3SdRqNCEZY6YjFp8wdx4qtzSDQ4HcrIZLvEHMaELarzqFWFb3v1ev19eeZlip2m4tOR1B3Hj6qpODKYEOEhMFE+fP1+vrzzROEwKNUwSI1TSinUCVMmRTIVQqomRlMhVGqVSVEyUqVUQQUQKCFUsqKVRYEWtyrvTknX73USkoJXAU5bkO7z9Pr6c8iXEc/T6+nPKpIkz33ZKVZFHXE4NGe8ncOKaKKvadg0ZnUncOPolmlvcAMhoAkJSkyYHf5SvfXgBkNAElNyLWkmgFSdFY4huAxOp9h++vLNSgTFgkcaYDPU+w/f7KxSFpDgaEbjT5jJIU7WYVOWg3/RKphdZ1qjnAE4o/4ZmjtEbntHepl6Lm26yPjoXULT3ZG4tI4HTkqi6uf/H0WiyWt7aga99jsWP/AFDR3H7N0sfZ9uf315cdJIguwMwbd989NFiRAW02Zj8Yqg5mEnH/ACOOYWUVqBTtZAbvBK6g5mbziLg/nkQHTkDKDmlquszBiTk0VJ+L8o5lFJaXAfyxk3M736oLYdrOzf2mtBjM68QLiwxPEgkWIQws6IQWuwT3H1OFQReHeZX428Rn7jNctxIEgd9/wgTC9H0mDTFZJpLxd/DNhuHCrrO5zSXnMYObgMTvGa5LOrewSSVIGFBUNaL10BrRkMR5rTKCbC6MkF0NoD946uaOgIO4uAPiqNiMvwzx60qPEYfNqv8ACaDKZ3IDvK7dkSB+oQDgZ+6r8PbvktdwLo5i7gekQnbtSFmEcGO8kN8cK1VU22pDQgMABxoCTTXNc6LKh+o+ikjSM8/VXO2/aHYdHIW/Zav6mobTHSwXX2lG6KUPZlQV/wApw+dKHeAj0sobQ6RvdlayUf8AsAJ+YKG0ZiYYZBnS6dx0cDwqxHaB6yCFwFA0OYTndLT2QTuxp5KzbG71dzxxAcPjH1VVejG0B7cQfk7dMfMWXNjcHC640Iwa45cGuO7cdOWVbmkEgihGYKrWpjw4BrjQjBrju0a7huOmWWWL3enfZQwkhkpxBzGhH3qmljAxbi05HUHceKqc0g0IoRmE8UtOIOY0IVnNGOISps+fr9UZGUxGIPnyPFKnCcGcJqpVK1RTplAUyREIqJgpVRBFRRGqCikoqFRRAqFBElFo1KjWncnoa8PulEpKiU7jzBQrTn6fVKTRBKggrIo64uwaMzqTuHFCNlcTgB58hxQklJ4AZDQJSkJJsFJZa8AMhoAkawkgAVJyAUa0k0AqTkF09n2EvJa1zQAP5khIAAPwMJzrv1x0zrc6LKYsFVFYnljnMaXBpDXyCl0V+EHdvdllpnjnhLSMQQRUEZEcK8ajwXorS59zqY3xiIDtXX5/qpkM644riSMye81GQbShNNABk1WVaBpZMzx1771VW7UYTvxE2jJ7vy5LOyMAXnZaDU/TiqpHEmp++CaWQuNT9ANABoEjWk4BZ0w1KAFUS/d5ovIyGW/f9FUoHEIp7+uu/wDdaRbXUxALqUEnxeO9ZUqtp7RUp+6Y7z1HA5GQUbqBRRRUoKIoIqKLtbGmL45YCKl0Ru82G+B51I5lV9H5rshNRiPm01/dZtk2i5NG78wHgcPdGbsPJHea4g7syA7x9eYWzZnezNOrwDr9DB+rk9Mezio3X8+t1G7NmJc1rDQOIBNAKVwoXFbY9hy3aPewDSlXEe1PFYXbRlIwfTeBTzWbtv8AxS+bkf8AiB0MDncpDR8IkpwaP+JJ0n7Lp2kNbF1fWA0xGnxXskljkIgkH4XtdlUEO7DhTUUohFYX3e20AaXy1t39WNacP+CGhrC6/KHF0ZbRrXEY4gg5LU4PeQajNxu6W3MWgx7xE54XTtne84ixHwgxlV2gBwvtywBGZB0rvG4+HPMF1oLPK2ymYWfs9Ya2gk0pda0xXa0Iq4HLXgVgljBF5mWo1afcbiuRTqh8wZgkZGRwMTB5G+oWUOvfvl3nrkhwcKHMYNdw0a72OmWWVRBGBzShWh1cDnofY+x9srhZEeXogx9OWo3oubqMvvNIQo1yYJ0ylUConTIqKVRRlRALTZ7HLJ3I3OG8A088l1uj+ymOHXTCra9lujiMyeFdNfXrW7aoAoMAMgMAOQVD68GAEhfFgvJz2OVneY4cae6oW+3bVJyK5geSalNTqz7wUa7VMU4akTnHHz5cFaVYjJXP7CQuUc6qVBBFFrdTl95JQg5yBQKZ768tBuSgE4DNCisvUFBmcz7D3PtmhSm1gmvXRQZnAu4fhb7nXLLNrBZDI6hcGtAq97u6xupO/cAMSSAFmKviFG1dWlahuV52XkN/kq3zundMHXPpxSGwXStVsiawNiaWxA4E4SSuGbjTuj0wA1K40spcan6AaADQIyylxqfoBoANAka0k0CrYwMHZPfdhZQDiVGtJNAi9wpQZanf9Pvk0jwBdblqd/04fYpJTFTN1s2TLE2aN0zL8YcC9mJvNGYwcPXzyI2lJE6RzomXGGlxmJoAAMSSa5VzxXoehPQmS2/zZHGOzg0v0F55GYjBww1ccNMcafTrP0Q2ZE2jbLG/e6Qda48avrTwouHtXi9ChXLWgvfYEA2Fyb3jeveATgYss9XaWU8r4Igvtm1eiWzpQR1DYzo6GkZHgOyfEFfMOk3R2Syvzvxu7j6U39lw+F3r506OzbUKzZ3SOqo2fxOhWf7MGHaa9FwlEUFqXQUUUUUUROS6NqtEBe6Q9aS44gUa3KhG9c1FXUqxpggAGYNxOJ+5TtcR33qt4tFP8OGMDO84mSni7JCW3yEUD3cadnypp81i0TMzCc7VViA4gaDyj5NAHooajose/glIBxIVrHgi67LQ7vp98635nmfVBZ4VcStr7bMI+oLz1YJdcwu1NDXjkMVnjkLTUfQjcUT3B4qpEC3feboi4gq97R3m5ajd9OKRPZe8PvRVJwoNFbWufgfYpFE25MmQCKVFNKMplEqJRRXctO0g1oaDg0ADwC409qc9VSd5WMWVrVXCVkS0xxIxrTGr2tCcBYJBQkJKp7V3jzVYToooIKIEqEqFBRHeggmrTLxPsEiiVKgrWNHedloN/wBOKSSQuNT9ANwTWnvFVJUoE3Ra0k0CZ7wBdblqd/0++R+A/qHoVUlQygr7FZDJJHEMDI9jG8C9waCfNUx5jmPVdTor/wBdZ/8AzR/1BVVnFrCRof2VdZ5axzhwBPovulmYyGNkUYoxjQ1o4Aa7zrXis81pUmWGVcfYPDaTRJuV4ertFSs/dJUmtC4+24WzRPid8Qw4O+E+BotUyxSr0tDZm4XV2DYmAgjK+TEKKy1d9/6nepRWQr06/9k="
                alt=""
                className="absolute inset-0 -z-10 h-full w-full object-cover object-right md:object-center"
            />
            <div
                className="hidden sm:absolute sm:-top-10 sm:right-1/2 sm:-z-10 sm:mr-10 sm:block sm:transform-gpu sm:blur-3xl"
                aria-hidden="true"
            >
                <div
                    className="aspect-[1097/845] w-[68.5625rem] bg-gradient-to-tr from-[#ff4694] to-[#776fff] opacity-20"
                    style={{
                        clipPath:
                            'polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%)',
                    }}
                />
            </div>
            <div
                className="absolute -top-52 left-1/2 -z-10 -translate-x-1/2 transform-gpu blur-3xl sm:top-[-28rem] sm:ml-16 sm:translate-x-0 sm:transform-gpu"
                aria-hidden="true"
            >
                <div
                    className="aspect-[1097/845] w-[68.5625rem] bg-gradient-to-tr from-[#ff4694] to-[#776fff] opacity-20"
                    style={{
                        clipPath:
                            'polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%)',
                    }}
                />
            </div>
            <div className="mx-auto max-w-7xl px-6 lg:px-8">
                <div className="mx-auto max-w-2xl lg:mx-0">
                    <h2 className="text-4xl font-bold tracking-tight text-white sm:text-6xl">
                        ¡Bienvenidos a GIPHY CLON!
                    </h2>
                    <p className="mt-6 text-lg leading-8 text-gray-300 text-justify">
                        Un proyecto creado por un apasionado estudiante de
                        frontend. Descubre un mundo de diversión y expresión con
                        nuestro clon de Giphy. Explora una amplia gama de gifs,
                        encuentra la dosis perfecta de entretenimiento y
                        comparte momentos divertidos con tus amigos. ¡Déjate
                        llevar por la creatividad y la diversión con nuestra
                        extensa colección de gifs!
                    </p>
                </div>
            </div>
        </div>
    );
};
