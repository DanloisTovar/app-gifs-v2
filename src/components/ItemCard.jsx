import PropTypes from 'prop-types';
export const ItemCard = ({ id, title, url }) => {
	return (
		<>
			<div
				key={id}
				className="w-full sm:w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 max-w-sm rounded overflow-hidden shadow-lg m-4 flex flex-col"
				/* style={{ height: '450px' }} tamaÃ±o de tarjeta*/
			>
				<div className="w-full h-64 overflow-hidden">
					<img
						className="w-full h-full object-cover"
						src={url}
						alt="Imagen de la tarjeta"
					/>
				</div>
				<div className="px-6 py-4 flex-grow">
					<div className="font-bold text-xl mb-2">{title}</div>
				</div>
			</div>
		</>
	);
};
ItemCard.propTypes = {
	id: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
