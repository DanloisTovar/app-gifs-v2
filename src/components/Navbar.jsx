import { Disclosure } from '@headlessui/react';

// imagenes:
import logoGiphy from '../assets/img/giphy_logo.png';

export const Navbar = () => {
	return (
		<Disclosure as="nav" className="bg-gray-400">
			{() => (
				<>
					<div className="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8">
						<div className="relative flex h-16 items-center justify-between">
							<div className="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start">
								<div className="flex flex-shrink-0 items-center">
									<img
										className="h-24 w-auto"
										src={logoGiphy}
									/>
								</div>
							</div>
						</div>
					</div>
				</>
			)}
		</Disclosure>
	);
};
