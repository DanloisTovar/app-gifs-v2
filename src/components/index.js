export * from './Cards';
export * from './Footer';
export * from './Formulario';
export * from './Header';
export * from './ItemCard';
export * from './Navbar';
export * from './Spinner';
