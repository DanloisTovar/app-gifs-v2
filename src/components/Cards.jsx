import PropTypes from 'prop-types';
import '../assets/styles/card.css';
import { ItemCard } from './ItemCard';
import { UseFetch } from '../hooks/UseFetch';
import { Spinner } from './Spinner';
import { useEffect, useState } from 'react';

export const Cards = ({ onAddInfoCategoriesForCard }) => {
	// hooks:
	const { gifts, isLoading } = UseFetch(onAddInfoCategoriesForCard);
	const [showTimeoutMessage, setShowTimeoutMessage] = useState(false);
	useEffect(() => {
		const timeoutId = setTimeout(() => {
			setShowTimeoutMessage(true);
		}, 10000); // 30 segundos

		// Limpia el temporizador si el componente se desmonta antes de que se agote el tiempo
		return () => clearTimeout(timeoutId);
	}, []);

	return (
		<>
			{isLoading ? (
				<>
					{showTimeoutMessage ? (
						<p className="mt-4 text-indigo-500 font-bold text-center">
							No hay conexi√≥n con la API
						</p>
					) : (
						<Spinner />
					)}
				</>
			) : (
				<>
					<div className="flex justify-center items-center mb-4 min-h-0">
						<h1 className="text-indigo-500 text-5xl font-bold">
							{onAddInfoCategoriesForCard}
						</h1>
					</div>

					<div className="flex flex-wrap justify-around">
						{gifts.map((gif) => (
							<ItemCard
								key={gif.id}
								{...gif} /* exparsion de props */
							/>
						))}
					</div>
				</>
			)}
		</>
	);
};

Cards.propTypes = {
	onAddInfoCategoriesForCard: PropTypes.string.isRequired,
};
