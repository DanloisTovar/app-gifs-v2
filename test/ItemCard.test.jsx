import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { ItemCard } from '../src/components'; // Asegúrate de importar el componente adecuadamente
/* Proptypes: */
describe('ItemCard component PropTypes', () => {
	test('verifica las PropTypes de ItemCard', () => {
		const consoleErrorSpy = jest
			.spyOn(console, 'error')
			.mockImplementation(() => {});

		const propTypesErrorId = PropTypes.checkPropTypes(
			ItemCard.propTypes,
			{ id: '123', title: 'Title', url: 'https://example.com/image.jpg' },
			// Valores simulados para las pruebas
			'prop',
			'id',
		);

		const propTypesErrorTitle = PropTypes.checkPropTypes(
			ItemCard.propTypes,
			{ id: '123', title: 'Title', url: 'https://example.com/image.jpg' }, // Valores simulados para las pruebas
			'prop',
			'title',
		);

		const propTypesErrorUrl = PropTypes.checkPropTypes(
			ItemCard.propTypes,
			{ id: '123', title: 'Title', url: 'https://example.com/image.jpg' }, // Valores simulados para las pruebas
			'prop',
			'url',
		);

		expect(propTypesErrorId).toBeUndefined();
		expect(propTypesErrorTitle).toBeUndefined();
		expect(propTypesErrorUrl).toBeUndefined();
		expect(consoleErrorSpy).not.toHaveBeenCalled();
	});
});

/* Snapshot: */
describe('ItemCard component Snapshot', () => {
	test('snapshot del componente ItemCard coincide con la última versión', () => {
		const { asFragment } = render(
			<ItemCard
				id="123"
				title="Title"
				url="https://example.com/image.jpg"
			/>,
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
