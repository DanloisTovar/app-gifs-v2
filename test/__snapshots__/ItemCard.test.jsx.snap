// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard component Snapshot snapshot del componente ItemCard coincide con la última versión 1`] = `
<DocumentFragment>
  <div
    class="w-full sm:w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 max-w-sm rounded overflow-hidden shadow-lg m-4 flex flex-col"
  >
    <div
      class="w-full h-64 overflow-hidden"
    >
      <img
        alt="Imagen de la tarjeta"
        class="w-full h-full object-cover"
        src="https://example.com/image.jpg"
      />
    </div>
    <div
      class="px-6 py-4 flex-grow"
    >
      <div
        class="font-bold text-xl mb-2"
      >
        Title
      </div>
    </div>
  </div>
</DocumentFragment>
`;
